sdb/commands/print.py

```diff
@@ -0,0 +1,100 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+
+import drgn
+import sdb
+
+
+class Print(sdb.SingleInputCommand):
+    """
+    Format output for shell script consumption.
+
+    EXAMPLES
+        By default getting the address of a global variable
+        looks like this:
+
+            sdb> addr spa_namespace_avl
+            (avl_tree_t *)spa_namespace_avl+0x0 = 0xffffffffc0954fe0
+
+        One could use the `print` command to format the above
+        to something that is better for consumption like this:
+
+            sdb> addr spa_namespace_avl | print -nr
+            0xffffffffc0954fe0
+    """
+
+    names = ["print"]
+
+    @classmethod
+    def _init_parser(cls, name: str) -> argparse.ArgumentParser:
+        #
+        # Check the following for available options that are built-in to drgn:
+        # drgn.readthedocs.io/en/latest/api_reference.html#drgn.Object.format_
+        #
+        parser = super()._init_parser(name)
+        parser.add_argument("-c",
+                            "--char",
+                            action="store_true",
+                            default=False,
+                            help="print character types as character literals")
+        parser.add_argument("-d",
+                            "--dereference",
+                            action="store_true",
+                            default=False,
+                            help="print type name only")
+        parser.add_argument("-n",
+                            "--nosymbolize",
+                            action="store_true",
+                            default=False,
+                            help="skip symbolizing")
+        parser.add_argument("-s",
+                            "--sameline",
+                            action="store_true",
+                            default=False,
+                            help="put all output in a single line")
+        parser.add_argument("-r",
+                            "--raw",
+                            action="store_true",
+                            default=False,
+                            help="don't print type names")
+        parser.add_argument(
+            "-R",
+            "--RAW",
+            action="store_true",
+            default=False,
+            help=("don't symbolize and don't print type names,"
+                  " member names in structs, nor indices in arrays"))
+        return parser
+
+    def _call_one(self, obj: drgn.Object) -> None:
+        if self.args.RAW:
+            self.args.raw = True
+            self.args.nosymbolize = True
```

Comment:
[nit] It seems a little unconventional to be changing `self.args.*` for each object.  The simplest way around this would be to have local variables, like `raw = self.args.raw or self.args.RAW`.

Another option would be to use a custom Action subclass that would store True in all 3 members.  See https://docs.python.org/3/library/argparse.html#argparse.Action

Reply:
I updated this and rebased with the latest array changes - let me know what you think